MapboxCoreNavigation/RouteController.swift

```diff
@@ -780,10 +806,17 @@ struct SessionState {
     var originalRoute: Route
 
     var pastLocations = FixedLengthQueue<CLLocation>(length: 40)
+    
+    var timeInForeground: TimeInterval
+    var timeInPortrait: TimeInterval
+    var previousForegroundChange: Date?
+    var previousOrientationChangeToPortrait: Date?
 
     init(currentRoute: Route, originalRoute: Route) {
         self.currentRoute = currentRoute
         self.originalRoute = originalRoute
+        self.timeInForeground = 0
```

Comment:
Nit: these can be moved to the property declaration:
`var timeInForeground: TimeInterval = 0`